README.md

```diff
@@ -124,6 +124,38 @@ To manually run just the build task use `yarn run build`.
 
 _If you need the smart contracts ABI files for dApp integrations, use `yarn run abi` to get a complete list of all ABI's inside of the `abi` directory._
 
+### Development with live interaction & manual testing (supports Metamask)
```

Comment:
I do not think we should be instructing users to use Ganache / Remix. Currently we're using hardhat for deployment and testing

Reply:
It is very difficult to develop new contracts in this repo and to make the whole workflow working with deployment script and tests on first try.

The way I developed the Faucet contract was:

1) First develop the generic faucet contract using Remix and Ganache GUI.
2) Test and verify that the expected AC is met
3) After I had working Faucet contract that met all requirements I started writing test cases from the actual test runs I did by hand.
4) After that I wrote the deployment script


I can't imagine working in reverse order. There is so many pieces that need to go together an the Development experience in this repo just isn't great.

First I need a contract file in order to get hardhat artifacts, ABIs and Typechain types.
Then I need to write a deployment script and then test case before I get to testing the contract itself.

The workflow above gave me so much flexibility. Suddenly I could change a variable in the contract and immediately see it working in Remix and I could quickly iterate on the fine details.

Without this, I would always had to adjust tests first in order to see how it changed.

This repo is more suited for TDD. Sure I could work this way. I didn't in this instance and this is the flow.

How do you develop with this repo? A training session would perhaps come handy.

@alexanderattar @ACC01ADE

Reply:
I'm mainly suggesting we don't include these instructions in the README because it's going to be confusing for auditors when they look at the code. It's fine to keep these notes for yourself or we could create a notion doc for internal use. We can discuss when @ACC01ADE does a run through the codebase with the whole team the best process for this moving forward

Reply:
OK, Notion sounds good

Reply:
OK, Notion sounds good

Reply:
We'll be working on more thorough documentation for devs at a later time, so this detailed explanations can be provided in the Notion doc as @alexanderattar suggested. I do like the idea of keeping the current README as minimal and to the point as possible for just cloning, building, running the thing and nothing else. We can provide a link at the top later that sends developers to a dedicated Notion section that goes into more details about the contracts, building, and interacting with them.